packages/react-components/react-aria/src/activedescendant/useActiveDescendant.ts

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { isHTMLElement } from '@fluentui/react-utilities';
+import { useOptionWalker } from './useOptionWalker';
+import { ActiveDescendantOptions } from './types';
+import { ACTIVEDESCENDANT_ATTRIBUTE } from './constants';
+
+export function useActiveDescendant<TActiveParentElement extends HTMLElement, TListboxElement extends HTMLElement>(
+  options: ActiveDescendantOptions,
+) {
+  const { imperativeRef, matchOption } = options;
+  const activeParentRef = React.useRef<TActiveParentElement>(null);
+  const { listboxRef, optionWalker } = useOptionWalker<TListboxElement>({ matchOption });
+  const getActiveDescendant = () => {
+    return listboxRef.current?.querySelector(`[${ACTIVEDESCENDANT_ATTRIBUTE}]`);
+  };
+
+  const setActiveDescendant = (nextActive: HTMLElement | undefined) => {
+    const active = getActiveDescendant();
+    if (isHTMLElement(active)) {
+      active.removeAttribute(ACTIVEDESCENDANT_ATTRIBUTE);
+    }
+
+    if (nextActive) {
+      nextActive.setAttribute(ACTIVEDESCENDANT_ATTRIBUTE, '');
+      activeParentRef.current?.setAttribute('aria-activedescendant', nextActive.id);
+    } else {
+      activeParentRef.current?.removeAttribute('aria-activedescendant');
+    }
+  };
+
+  React.useImperativeHandle(imperativeRef, () => ({
+    first: () => {
+      if (!listboxRef.current || !activeParentRef.current) {
+        return;
+      }
+
+      const first = optionWalker.first();
+
+      if (!first || !isHTMLElement(first)) {
+        return;
+      }
+
+      setActiveDescendant(first);
```

Comment:
If you will do proposed changes in `useOptionWalker`, we can simplify to:

```suggestion
      if (!listboxRef.current || !activeParentRef.current) {
        return;
      }

      const first = optionWalker.first();

      if (first) {
        setActiveDescendant(first);
      }
```